index.bs

```diff
@@ -2367,6 +2379,43 @@ To <dfn>render viewport to a canvas</dfn> given |viewport| and |rect|:
 
 </div>
 
+<div algorithm>
+To <dfn>encode a canvas as Base64</dfn> given |canvas| and |format|:
+
+1. If the |canvas|’s bitmap’s [=origin-clean=] flag is set to false, return
+   [=error=] with error code [=unable to capture screen=].
+
+1. If the |canvas|’s bitmap has no pixels (i.e. either its horizontal dimension
+   or vertical dimension is zero), return [=error=] with error code [=unable to
+   capture screen=].
+
+1. Let |quality| be [=undefined=].
+
+1. Let |type| be "image/png".
+
+1. If |format| is "webp", set |type| to "image/webp".
+
+1. Otherwise, if |format| matches the <code>browsingContext.JpegFormat</code>
+   production, set |type| to "image/jpeg" and let |quality| be the
+   <code>quality</code> field of |format|.
+
+1. Otherwise, if |format| is not <code>"png"</code>, return [=error=] with error
+   code [=invalid argument=].
+
+1. Let |file| be [=a serialization of the bitmap as a file|a serialization of
+   the |canvas| element’s bitmap as a file=] with |type| and |quality|.
+
+1. Let |data url| be a [=data url|<code>data:</code> URL=] representing |file|.
+
+1. Let |index| be the [=String.prototype.indexOf|index of=] "," in |data url|.
```

Comment:
We should not define it in terms of ECMAScript very likely.

Reply:
Could you provide an alternative?

Reply:
I believe there should be something in https://infra.spec.whatwg.org

Reply:
e.g., https://infra.spec.whatwg.org/#strictly-split

Reply:
It seems infra spec only defined [substring](https://infra.spec.whatwg.org/#code-point-substring-to-the-end-of-the-string) but not _index of_.

I don't have a strong opinion here, but the current text is derived from WebDriver which uses ECMAScript to define the operations described here. Also, several parts of WebDriver and WebDriver BiDi use ECMAScript operations to define operations in algorithms.

@jgraham @gsnedders Is the terminology here fine to use? If there is push-back to the terms here, I'd appreciate some guidance on how to move forward. I think https://infra.spec.whatwg.org/ defines _some_ algorithms, but not all of them. It's also important we do not introduce additional references as WebDriver BiDi already references ECMAScript. A priori, any new reference we introduce here should be migrated to (before or after this PR).

Again, not strong opinions here, so I'm okay with what exists here.

Reply:
I think these days we can bypass the entire issue by just using `[=forgiving base64 encode=] |file|'s data` or something, and bypass the entire "data URL" thing.

Reply:
Resolved.